packages/algob/src/internal/deployer.ts

```diff
@@ -489,7 +497,87 @@ class DeployerBasicMode {
 	async getReceiptTxns(txns: Transaction[]): Promise<TxnReceipt[]> {
 		return await this.algoOp.getReceiptTxns(txns);
 	}
+
+	/**
+	 * Creates an algosdk.Transaction object based on execParams and suggestedParams
+	 * @param execParams execParams containing all txn info
+	 * @param txParams suggestedParams object
+	 * @returns array of algosdk.Transaction objects
+	 */
+	makeTx(execParams: wtypes.ExecParams[], txParams: algosdk.SuggestedParams): Transaction[] {
+		const txns: Transaction[] = [];
+		for (const [_, txn] of execParams.entries()) {
+			txns.push(mkTransaction(txn, txParams));
+		}
+		return txns;
+	}
+
+	/**
+	 * Signes a Transaction object with the signer
+	 * @param transaction transaction object.
+	 * @param signer object that signes the transaction
+	 * @returns SignedTransaction
+	 */
+	async signTx(
+		transaction: algosdk.Transaction,
+		signer: wtypes.Sign
+	): Promise<SignedTransaction> {
+		let decodedResult: Uint8Array;
+		switch (signer.sign) {
+			case wtypes.SignType.SecretKey: {
+				decodedResult = transaction.signTxn(signer.fromAccount.sk);
+				break;
+			}
+			case wtypes.SignType.LogicSignature: {
+				signer.lsig.lsig.args = signer.args ?? [];
+				decodedResult = algosdk.signLogicSigTransactionObject(transaction, signer.lsig).blob;
+				break;
+			}
+			default: {
+				throw new Error("Unknown type of signature");
+			}
+		}
+		return algosdk.decodeSignedTransaction(decodedResult);
+	}
+
+	/**
+	 * Creates an algosdk.Transaction object based on execParams and suggestedParams
+	 * and signs it using provided signer
+	 * @param execParams execParams containing all txn info
+	 * @param txParams suggestedParams object
+	 * @param signer object that signes the transaction
+	 * @returns array of algosdk.SignedTransaction objects
+	 */
+	async makeAndSignTx(
+		execParams: wtypes.ExecParams[],
+		txParams: algosdk.SuggestedParams,
+		signer: wtypes.Sign
+	): Promise<SignedTransaction[]> {
+		const signedTxns: SignedTransaction[] = [];
+		const txns: Transaction[] = this.makeTx(execParams, txParams);
+		txns.forEach(async (txn) => signedTxns.push(await this.signTx(txn, signer)));
+		return signedTxns;
+	}
+
+	/**
+	 * Sends signedTransaction and waits for the response
+	 * @param transactions array of signedTransaction objects.
+	 * @param rounds number of rounds to wait for response
+	 * @returns ConfirmedTxInfo
+	 */
+	async sendTxAndWait(
```

Comment:
you don't need to mark function as `async` if the only `await` you have in return. Just keep return type as `Promise` and remove async / await.